### PR TITLE
[FAB-18041] Add Node.js to CC as external service

### PR DESCRIPTION
**Pull request by shimos**
#### Type of change

- Documentation update

#### Description

As FABCN-406 is merged, this patch adds description for Node.js chaincode as an external
service.

#### Related issues

https://jira.hyperledger.org/browse/FAB-18041

### CODE REVIEW

**Review by pamandrejko — DISMISSED** [review]
Documentation changes look reasonable to me. I'll the maintainers weigh on the technical content.

File: docs/source/cc_service.md [hunk 1]
```diff
@@ -199,7 +199,11 @@ exit 1
 
 ## Writing chaincode to run as an external service
 
-Currently, the chaincode as an external service model is only supported by GO chaincode shim. In Fabric v2.0, the GO shim API adds a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:
+Currently, the chaincode as an external service model is supported by GO chaincode shim and Node.js chaincode shim.
```
Comment (mastersingh24) [inline docs/source/cc_service.md]:
```suggestion
Currently, the chaincode as an external service model is supported by Go chaincode shim and Node.js chaincode shim.
```

File: docs/source/cc_service.md [hunk 2]
```diff
@@ -199,7 +199,11 @@ exit 1
...
+
+### GO
```
Comment (mastersingh24) [inline docs/source/cc_service.md]:
```suggestion
### Go
```

File: docs/source/cc_service.md [hunk 3]
```diff
@@ -199,7 +199,11 @@ exit 1
...
+
+In Fabric v2.0, the GO shim API adds a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:
```
Comment (mastersingh24) [inline docs/source/cc_service.md]:
```suggestion
In Fabric v2.0, the Go shim API provides a `ChaincodeServer` type that developers should use to create a chaincode server.  The `Invoke` and `Query` APIs are unaffected. Developers should write to the `shim.ChaincodeServer` API, then build the chaincode and run it in the external environment of choice. Here is a simple sample chaincode program to illustrate the pattern:
```

File: docs/source/cc_service.md [hunk 4]
```diff
@@ -253,9 +257,54 @@ The key to running the chaincode as an external service is the use of `shim.Chai
 
 Then build the chaincode as suitable to your GO environment.
```
Comment (mastersingh24) [inline docs/source/cc_service.md]:
```suggestion
Then build the chaincode as suitable to your Go environment.
```

File: docs/source/cc_service.md [hunk 5]
```diff
@@ -253,9 +257,54 @@ The key to running the chaincode as an external service is the use of `shim.Chai
...
 
+### Node.js
+
+`fabric-shim` package for Node.js chaincode adds `shim.server` API to run the chaincode as an external service. If you are using contract APIs, you may want to use the `server` command added to `fabric-chaincode-node` CLI to run a contract in the external service mode.
```
Comment (mastersingh24) [inline docs/source/cc_service.md]:
```suggestion
`fabric-shim` package for Node.js chaincode provides the `shim.server` API to run chaincode as an external service. If you are using contract APIs, you may want to use the `server` command provided by the `fabric-chaincode-node` CLI to run a contract in the external service mode.
```

**Review by mastersingh24 — DISMISSED** [review]

**Comment by shimos** [comment]
@mastersingh24  Thanks for the review. I have fixed the patch as per your suggestions.

**Review by lindluni — APPROVED** [review]

**Pull request merged**